app/controllers/claims_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ClaimsController < ApplicationController
+  def new
+    contribution = Listing.find(params[:contribution_id])
+    if contribution.person&.email.blank?
+      flash.now[:alert] = "We are sorry, this contributor hasn't provided
+                         an email address yet and can't communicate this way".squish
+    end
+    render locals: { contribution: contribution }
+  end
+
+  def create
+    # TODO: Need to handle race conditions to prevent creating multiple matches for same contribution.
+
+    contribution = Listing.find(params[:contribution_id])
```

Comment:
Another fixme comment please to extract an interactor 🙏🏾.
In this case to keep controllers focussed.